pybamm/experiment/step/base_step.py

```diff
@@ -71,6 +71,8 @@ def __init__(
         description=None,
         direction=None,
     ):
+        self.input_duration = duration
+        self.input_value = value
```

Comment:
If these are only used for the copy method, is there a strong reason not to just use the python copy.deepcopy() method? For the array items the constructor is looping over the arrays anyway, so I would assume python's built in copy would be safer, cleaner and potentially faster than writing our own copy operation.

Reply:
In general I've found deepcopy to be a huge performance bottleneck, and there's a few attributes of step that we don't want to copy (start time and end time, which are calculated based on the adjacent steps). The proper fix willl be to refactor steps so we don't have to copy at all, but that is outside the scope of this PR